common/crypto.hpp

```diff
@@ -37,6 +37,22 @@ enum CryptoErrorCode {
 	VerificationError,
 };
 
+class PrivateKey;
+using ExpectedPrivateKey = expected::expected<PrivateKey, error::Error>;
+
+class PrivateKey {
+public:
+	static ExpectedPrivateKey LoadFromPEM(const string &private_key_path, const string &passphrase);
```

Comment:
Nice! 

But why `static`?

Reply:
Isn't it because it's a factory function?

Reply:
But is it needed? :thinking:

Reply:
Constructors can't return errors (without exceptions), so to me it looks like it's needed. But feel free to correct me, @larsewi, I'm just guessing at your motivation.